### PR TITLE
Add tor

### PR DESCRIPTION
**Pull request by CRKatri**

### CODE REVIEW

File: tor.mk [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+DEB_TOR_V    ?= $(TOR_VERSION)
+
+tor-setup: setup
+	$(call EXTRACT_TAR,tor-$(TOR_VERSION).tar.gz,tor-$(TOR_VERSION),tor)
+
+ifneq ($(wildcard $(BUILD_WORK)/tor/.build_complete),)
+tor:
+	@echo "Using previously built tor."
+else
+tor: tor-setup
```
Comment (Diatrus) [inline tor.mk:17]: Does tor link to openssl, libevent, xz, and/or zstd? If so, why aren't they a required target here, and if not, why are they listed as Depends in the control file?

File: tor.mk [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		LFLAGS2="$(CFLAGS)"
+	+$(MAKE) -C $(BUILD_WORK)/tor install
+	touch $(BUILD_WORK)/tor/.build_complete
+endif
+
+tor-package: tor-stage
+	# tor.mk Package Structure
+	rm -rf $(BUILD_DIST)/tor
+	mkdir -p $(BUILD_DIST)/tor
+	mkdir -p $(BUILD_DIST)/tor/usr
```
Comment (Diatrus) [inline tor.mk:34]: The first mkdir command is useless. Just include `mkdir -p $(BUILD_DIST)/tor/usr` instead of both lines.

File: tor.mk [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+tor-setup: setup
+	$(call EXTRACT_TAR,tor-$(TOR_VERSION).tar.gz,tor-$(TOR_VERSION),tor)
+
+ifneq ($(wildcard $(BUILD_WORK)/tor/.build_complete),)
+tor:
+	@echo "Using previously built tor."
+else
+tor: tor-setup
+	cd $(BUILD_WORK)/tor && ./configure --prefix=$(BUILD_STAGE)/tor \
```
Comment (Diatrus) [inline tor.mk:18]: I have no doubt specifying `--prefix=/usr` would alleviate the issue you have during the package stage with everything installing to `$(BUILD_STAGE)/bin` and `$(BUILD_STAGE)/share`.

Comment (CRKatri) [reply tor.mk:18]: Setting the prefix to /usr will move /etc/tor to /usr/etc/tor I’ll look into how to fix that tonight

File: tor.mk [hunk 4]
```diff
@@ -30,13 +34,12 @@ endif
 tor-package: tor-stage
 	# tor.mk Package Structure
 	rm -rf $(BUILD_DIST)/tor
-	mkdir -p $(BUILD_DIST)/tor
+	mkdir -p $(BUILD_DIST)/tor/etc
 	mkdir -p $(BUILD_DIST)/tor/usr
```
Comment (Diatrus) [inline tor.mk:38]: You can get rid of all of these and just `cp -a $(BUILD_STAGE)/tor $(BUILD_DIST)`

**Comment by Diatrus** [comment]
Thanks for the contribution!

**Pull request merged**